src/lib/shim/src/lib.rs

```diff
@@ -36,6 +228,41 @@ extern crate shadow_shim_helper_rs;
 extern crate shadow_shmem;
 extern crate shadow_tsc;
 
+fn load() {
+    // First ensure thread local storage is initialized.
+    // TODO: Make the preferred mode configurable at runtime.
+    static TLS_INIT_DONE: LazyLock<()> = LazyLock::const_new(||
+        // SAFETY: LazyLock ensures this is only called once.
+        unsafe { tls::global_preferred_mode::set(tls::Mode::Native) });
+    TLS_INIT_DONE.force();
+
+    static STARTED_THREAD_INIT: ShimTlsVar<LazyLock<()>> =
+        ShimTlsVar::new(|| LazyLock::const_new(|| ()));
+    if STARTED_THREAD_INIT.get().initd() {
+        // Avoid recursion in initialization.
+        //
+        // TODO: This shouldn't be necessary once we've gotten rid of all
+        // calls to libc from the shim's initialization.
+        return;
+    }
+    STARTED_THREAD_INIT.get().force();
+
+    static THREAD_INIT: ShimTlsVar<LazyLock<()>> = ShimTlsVar::new(|| {
+        LazyLock::const_new(|| {
+            static GLOBAL_INIT: LazyLock<()> = LazyLock::const_new(|| ());
```

Comment:
Maybe add comments here as well to make the code easier to follow? For example that `THREAD_INIT` only allows the closure to run once per thread, and `GLOBAL_INIT` only allows the code in the if statement to run once per process?